src/App.js

```diff
@@ -65,12 +65,28 @@ export class ResourceEditor extends React.Component {
     }
   }
 
-  metadataHandler(resource) {
+  metadataHandler(fileResource) {
+    let resource = this.extendResource(fileResource)
     this.setState({
       resource,
     });
   }
 
+  /**
+   * Adds a new field bq_table_name to resources and formats name property
+   * @param {Object} resource 
+   */
+  extendResource(resource){
+    let newResource = {...resource}
+    newResource.title = resource.name
+    newResource.urlName = resource.name
+    
+    const newName = resource.name.split(".")[0].toUpperCase()
+    newResource.name = newName
```

Comment:
@risenW as I said make sure you can change the `resource.name` property.

and also make sure you can create and send new property as `resource.urlName`.